### PR TITLE
More flexible LIVD Lookup

### PR DESCRIPTION
**Pull request by RickHawesUSDS**
This PR adds a specialized `livdLookup` mapper for LIVD table lookups that uses more than just the `equipment_model_name` to lookup values in the table. In addition to the `equipment_model_name`, the mapper will use the `device_id` and the `equipment_uid`. This mapper is needed for HL7 ingestion because `equipment_model_name` is not available in the HL7 message. 

The `device_id` based lookup supports the GUDID and the [HHS Confluence guidance](fluence.hl7.org/display/OO/Proposed+HHS+ELR+Submission+Guidance+using+HL7+v2+Messages#ProposedHHSELRSubmissionGuidanceusingHL7v2Messages-DeviceIdentification). In particular, it parses the id in multiple ways and uses multiple columns to lookup. 

## Changes
- New livdLookup mapper and tests
- Covid-19 schema uses the new `livdLookup` mapper

### Testing
- [x] Tested locally?
- [x] Ran `quickTest all`?
- [x] Ran `./prime test` against local Docker ReportStream container?
- [ ] Downloaded a file from `http://localhost:7071/api/download`?
- [x] Added tests?

### Security
- [ ] Did you check for sensitive data, and remove any? 
- [ ] Does logging contain sensitive data?  
- [ ] Are additional approvals needed for this change?
- [ ] Are there potential vulnerabilities or licensing issues with any new dependencies introduced?

### Process
- [ ] Includes a summary of what a code reviewer should verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Security
*List potential security threats and mitigations in the PR*

## Fixes
*List GitHub issues this PR fixes*
- #1055

### CODE REVIEW

File: prime-router/src/main/kotlin/serializers/CsvSerializer.kt [hunk 1]
```diff
@@ -304,14 +304,9 @@ class CsvSerializer(val metadata: Metadata) {
...
             val valueNames = mapper.valueNames(element, args)
-            val valuesForMapper = valueNames.map { elementName ->
-                val valueElement = schema.findElement(elementName)
-                    ?: error(
-                        "Schema Error: Could not find element '$elementName' for mapper " +
-                            "'${mapper.name}' from '${element.name}'."
-                    )
-                val value = lookupValues[elementName]
-                    ?: error("Schema Error: No mapper input for $elementName")
+            val valuesForMapper = valueNames.mapNotNull { elementName ->
```
Comment (RickHawesUSDS) [inline prime-router/src/main/kotlin/serializers/CsvSerializer.kt:307]: This follows the mapper protocol used in translates.

File: prime-router/src/main/kotlin/serializers/Hl7Serializer.kt [hunk 2]
```diff
@@ -427,14 +427,9 @@ class Hl7Serializer(val metadata: Metadata) {
...
         }
-        val valuesForMapper = valueNames?.map { elementName ->
-            val valueElement = report.schema.findElement(elementName)
-                ?: error(
-                    "Schema Error: Could not find element '$elementName' for mapper " +
-                        "'${mapper.name}' from '${element.name}'."
-                )
-            val value = lookupValues[elementName]
-                ?: error("Schema Error: No mapper input for $elementName")
+        val valuesForMapper = valueNames?.mapNotNull { elementName ->
```
Comment (RickHawesUSDS) [inline prime-router/src/main/kotlin/serializers/Hl7Serializer.kt:430]: This follows the mapper protocol used in translates.

File: prime-router/src/main/kotlin/LookupTable.kt [hunk 3]
```diff
@@ -56,6 +57,26 @@ class LookupTable(
...
+        val lookupColIndex = headerIndex[lookupColumn.lowercase()] ?: return null
+        val findValue = if (ignoreCase) indexValue.lowercase() else indexValue
+        for (row in table) {
+            val rowsValue = if (ignoreCase) row[indexColIndex].lowercase() else row[indexColIndex]
+            if (rowsValue.startsWith(findValue)) return row[lookupColIndex]
+        }
+        return null
+    }
+
     /**
```
Comment (RickHawesUSDS) [inline prime-router/src/main/kotlin/LookupTable.kt:80]: Useful when lookups are using truncated values.

**Review by carlosfelix2 — APPROVED** [review]

**Comment by carlosfelix2** [comment]
@RickHawesUSDS  Merge from master to fix the build issue with the docs.

**Pull request merged**